### PR TITLE
Serve OpenAPI spec by default

### LINKED ISSUE #109: Provide default OpenAPI settings & definitions
Since #65, `custom-metrics-apiserver` serves an OpenAPI spec; however, it requires a configuration and the generation by the project using `custom-metrics-apiserver` of OpenAPI definitions using `openapi-gen`.

That leads to duplicated works, each implementation project having to generate its own OpenAPI definitions.

E.g.:
- `test-adapter` generates definitions [here](https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/9e9816d4c13e00e16fc48e72a1d7316dc878d507/Makefile#L16-L24) and configures OpenAPI [here](https://github.com/kubernetes-sigs/custom-metrics-apiserver/blob/9e9816d4c13e00e16fc48e72a1d7316dc878d507/test-adapter/main.go#L67-L69);
- `prometheus-adapter` generates definitions [here](https://github.com/kubernetes-sigs/prometheus-adapter/blob/268b2a8ec2ab29da82a5f08cc46df35bfd1555d4/Makefile#L94-L97) and configures OpenAPI [here](https://github.com/kubernetes-sigs/prometheus-adapter/blob/268b2a8ec2ab29da82a5f08cc46df35bfd1555d4/cmd/adapter/adapter.go#L290-L292);
- `keda` generates definitions [here](https://github.com/kedacore/keda/blob/main/Makefile#L126-L134) and configures OpenAPI [here](https://github.com/kedacore/keda/blob/f8ea5dd318b6a13dbdace872f82ac7c19c34b7c4/adapter/main.go#L201-L203).

It seams reasonable to provide these generated OpenAPI definitions in `custom-metrics-apiserver`, along with default OpenAPI serving settings (and let it possible to customize it).

Any thought?

/kind feature
/assign

### PR DESCRIPTION
**Pull request by olivierlemasle**
Resolves #109

---

This PR adds default OpenAPI definitions and settings, while keeping it customizable.

If the project using `custom-metrics-apiserver` continues to use

```go
cmd.OpenAPIConfig = genericapiserver.DefaultOpenAPIConfig(generatedopenapi.GetOpenAPIDefinitions, openapinamer.NewDefinitionNamer(apiserver.Scheme))
cmd.OpenAPIConfig.Info.Title = "name"
cmd.OpenAPIConfig.Info.Version = "1.0.0"
```
then nothing changes.

However, if an `OpenAPIConfig` is not provided, `custom-metrics-apiserver` will provide a default value, with its own generated definitions, which means that the OpenAPI spec will be served.

**Note 1:** I have split the definitions between custom_metrics, external_metrics and "core", but it is possible to use a single generated file.

**Note 2:** I could have let `test-adapter` use its custom OpenAPI settings to show how to use it. But I don't think it's really necessary (documentation could be added,  though).

### CODE REVIEW

File: pkg/cmd/builder_test.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package cmd
```
Comment (dgrisonnet) [inline pkg/cmd/builder_test.go:1]: nit: could you add the license here?

Side note, we could automate the update and verification of licenses at some point. We already did it in metrics-server at some point: https://github.com/kubernetes-sigs/metrics-server/blob/master/Makefile#L192-L208

**Comment by dgrisonnet** [comment]
I very like this approach, it is a great improvement that will remove a lot of tedious work for maintainers as I don't expect many projects to have a custom OpenAPI

File: pkg/cmd/builder_test.go (hunk 1 above)
Comment (olivierlemasle) [reply pkg/cmd/builder_test.go:17]: Oops, right. License added.

I'll add a license checker later.

File: Makefile [hunk 2]
```diff
@@ -9,20 +9,27 @@ VERSION?=latest
 .PHONY: all
 all: build-test-adapter
 
-.PHONY: build-test-adapter
-build-test-adapter: test-adapter/generated/openapi/zz_generated.openapi.go
-	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) go build -o $(OUT_DIR)/$(ARCH)/test-adapter sigs.k8s.io/custom-metrics-apiserver/test-adapter
+generated_openapis := core custom_metrics external_metrics
```
[the remark this answers is not in the document]
Comment (olivierlemasle) [reply Makefile]: Right, done.

**Pull request merged**